src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -245,22 +244,27 @@ public void InitializeService(ServiceHost serviceHost)
                                 return;
                             }
                             
-                            SubsetParams subsetRequestParams = new SubsetParams
-                            {
-                                OwnerUri = randomUri,
-                                BatchIndex = 0,
-                                ResultSetIndex = 0,
-                                RowsStartIndex = 0,
-                                RowsCount = Convert.ToInt32(rowCount)
-                            };
-                            // get the data to send back
-                            ResultSetSubset subset = await InterServiceResultSubset(subsetRequestParams);
                             SimpleExecuteResult result = new SimpleExecuteResult
                             {
-                                RowCount = query.Batches[0].ResultSets[0].RowCount,
+                                RowCount = rowCount,
                                 ColumnInfo = query.Batches[0].ResultSets[0].Columns,
-                                Rows = subset.Rows
+                                Rows = new DbCellValue[0][] 
```

Comment:
Why is this set to a empty object? if there are no rows, wouldn't we want this value to be null?

Reply:
It minimizes the risk that the caller will handle things incorrectly since this way you have a length for the rows no matter what. 

It's done this way in some other calls - basically it's easier to return an empty object since that's what this really is. It's a result set with no rows, not `null` rows.